lesson_02_homework.rb

```diff
@@ -1,46 +1,91 @@
-# Code Reading
 
+#def=Method, assigning to full_name 3 different variables -->
 def full_name(first_name, last_name, title)
-  # Example comment line
+  <!-- # Example comment line
 
+  #name assigned as nil -->
   name = nil
-
+  
+  <!-- #setting the first scenario/condition when running code,if it comes true then puts the tittle first then space, first name, space then last name. -->
   if title && first_name && last_name
     name = title + " " + first_name + " " + last_name
-  elsif title && last_name
+ # <!-- this send case is only when given title and last name -->
+ #  elsif title && last_name
     name = title + " " + last_name
+    # <!-- the third condition with first and last name -->
   elsif first_name && last_name
     name = first_name + " " + last_name
+    # <!-- only the first name -->
   elsif first_name
     name = first_name
+    # <!-- otherwhise just comes as this line of sentence telling that the name wasnt writen properly -->
   else
     raise "Oh no, that doesn't look like a name"
   end
-
+  # <!-- prints the correct statement that matched the code above -->
   return name
+  # <!-- closing the def method -->
 end
 
+
 # Fix Broken Code
 
+
 def full_name(f, l)
-  puts f + l
+  puts f + " " + l
 end
 
-puts full_nam("bubba", "chuck")
+puts full_name("bubba", "chuck")
 
 
 # Coding
 
+1
+  def add(num1,num2)
+   
+    total = num1 + num2
+    total
+ 
+
+  end 
+
+  puts add(1,2)
+
+
+
 
-## Enter your code for your 'add' method below
 
+2. Write a method, `join_strings` that does the following: 
+  - takes two string values as parameters and joins them together with a space in between, returning the result. 
+  - For example `join_strings("hello", "dolly")` should return `"hello dolly"`
 
 
+def join_strings(a, b)
+  puts a + " " + b
+end
+
+puts join_strings("Handsome", "Man")
+
+
+3. Write a method called 'old_enough_to_vote?' that does the following:
```

Comment:
:hammer: @loriksm This will cause the program to crash, it is not necessary, if you do want to keep this in, then it should be added as a comment